File: app/src/main/java/com/ahmedabdelmeged/pagingwithrxjava/PagingApp.java
package com.ahmedabdelmeged.pagingwithrxjava;

import android.app.Application;

import timber.log.Timber;

/**
 * Created by Ahmed Abd-Elmeged on 2/13/2018.
 */
public class PagingApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }
    }

}

File: app/src/main/java/com/ahmedabdelmeged/pagingwithrxjava/PagingApp.kt
package com.ahmedabdelmeged.pagingwithrxjava

import android.app.Application
import android.content.Context

import timber.log.Timber

/**
 * Created by Ahmed Abd-Elmeged on 2/13/2018.
 */
class PagingApp : Application() {

    companion object {

        lateinit var context: PagingApp

        @JvmStatic
        fun getContextExt(): Context = context.applicationContext

    }

    override fun onCreate() {
        super.onCreate()
        context = this
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
    }

}
